packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -1560,13 +1576,13 @@ class LayoutId extends ParentDataWidget<CustomMultiChildLayout> {
   /// An object representing the identity of this child.
   ///
   /// The [id] needs to be unique among the children that the
-  /// [CustomMultiChildLayout] manages.
+  /// [ParentType] manages.
   final Object id;
 
   @override
   void applyParentData(RenderObject renderObject) {
-    assert(renderObject.parentData is MultiChildLayoutParentData);
-    final MultiChildLayoutParentData parentData = renderObject.parentData;
+    assert(renderObject.parentData is ParentDataType);
+    final ParentDataType parentData = renderObject.parentData;
```

Comment:
we should probably revert this change since we don't need it anymore

Reply:
@Hixie you're saying literally revert back to MultiChildLayoutParentData? Or you're saying get rid of the assert? or get rid of the assert and the local parentData variable?

Reply:
I mean get rid of BaseLayoutId. Just pass in the children to _CupertinoDialogRenderWidget as two separate explicit slots.

Reply:
@Hixie do you not think BaseLayoutId is of general use? It seemed strange to me from the beginning that such tight coupling existed between LayoutId and CustomMultiChildLayout. Don't you think it's likely that we'll end up re-implementing this for the same reason down the road?